src/useEdgeAvoidingLayout/helpers.ts

```diff
@@ -0,0 +1,78 @@
+import { RefObject } from 'react'
+
+export interface Refs<
+  Target extends HTMLElement = HTMLElement,
+  Parent extends HTMLElement = HTMLElement,
+  Container extends HTMLElement = HTMLElement
+  > {
+  target: RefObject<Target>
+  parent: RefObject<Parent>
+  container: RefObject<Container>
+}
+
+export type Vertical = 'top' | 'bottom'
+export type Horizontal = 'left' | 'right'
+
+export interface AutoLayout {
+  vertical?: Vertical | 'auto'
+  horizontal?: Horizontal | 'auto'
+}
+
+export interface Layout extends AutoLayout {
+  horizontal: Horizontal
+  vertical: Vertical
+}
+
+export type BreakDistance = {
+  horizontal: number
+  vertical: number
+} | number
+
+const defaultLayout: AutoLayout = { vertical: 'auto', horizontal: 'auto' }
+
+export function getLayout (
+  { target, parent, container }: Refs,
+  breakDistance: BreakDistance,
+  layout: AutoLayout = {}
+): Layout {
+  const { vertical, horizontal } = { ...defaultLayout, ...layout }
+  const { top, left, height, width } = getMeasures(target.current, parent.current)
```

Comment:
Why not use the default argument for layout parameter?
`
  layout: AutoLayout = defaultLayout
  const { vertical, horizontal } = layout
`

Reply:
Because doing it like this allows you to only give a value for either horizontal or vertical and the other will default to auto. If you have a default value and you overwrite it with an object having only one of the two, the other will stay undefined. (Was my thought process at least, I haven’t checked if this is how it works)

Reply:
I understand!